data-prepper-plugins/saas-source-plugins/source-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/source_crawler/base/Crawler.java

```diff
@@ -39,15 +39,18 @@ public Crawler(CrawlerClient client, PluginMetrics pluginMetrics) {
     }
 
     public Instant crawl(LeaderPartition leaderPartition,
-                         EnhancedSourceCoordinator coordinator, int batchSize) {
+                         EnhancedSourceCoordinator coordinator, int batchSize, Duration pollingTimezoneOffset) {
         long startTime = System.currentTimeMillis();
         Instant lastLeaderSavedInstant = Instant.now();
         LeaderProgressState leaderProgressState = leaderPartition.getProgressState().get();
+        // Leader state is always saved in UTC since the timestamps in the api response are always in UTC
         Instant lastPollTime = leaderProgressState.getLastPollTime();
-        client.setLastPollTime(lastPollTime);
+        // Adjust the time specific to the crawling source service setting before polling for changes
+        Instant lastPollTimeWithOffsetAdjustment = lastPollTime.plusSeconds(pollingTimezoneOffset.toSeconds());
+        client.setLastPollTime(lastPollTimeWithOffsetAdjustment);
         Iterator<ItemInfo> itemInfoIterator = client.listItems();
         Instant latestModifiedTime = Instant.from(lastPollTime);
```

Comment:
No need to use Instant.from?
```suggestion
        Instant latestModifiedTime = lastPollTime;
```

Reply:
Addressed it